src/generate/name.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import _ from "lodash";
+
+/**
+ * Stores a name with common transformations cached for use in templates and file paths.
+ */
+export default class Name {
```

Comment:
Do we expect people to be modifying names all that often? I think it's probably fine to just use normal properties.

Reply:
This makes it so the transformations stay in sync with the original field